### PR TITLE
Breakable Stuff 1: ATMs, Chem dispensers, Shrubs

### PR DESCRIPTION
**Pull request by Flaborized**
## About the PR 

Lets players smack ATMs, chemical dispensers and shrubs with weapons to break them. Shrubs leave behind a few leaves that can be cleaned up with space cleaner, and ATMs spit out money. You can also shoot an ATM with any lethal ammo to break it, just like in the movies! Chem dispensers have a fair bit of health, so breaking all the dispensers in chemistry will take a while unless you have a saber or something. Numbers for all their health can be tweaked if necessary.

Also adds slightly better visual clarity to hitting fabricators, they now use attack_particle and hit_twitch, which should hopefully make it more clear you're actually doing something.

## Why's this needed?

I want there to be more breakable stuff in general. It feels very bad for things to be static and untouchable when they shouldn't be, and fails to sell that they are what they really are. I also broadly want traitors to eventually have more things to target that aren't just people, and more ways that aren't bombs to destroy them.

I also want to be able to remove shrubs for building purposes (they often get in the way) and I think players feel similarly often? And I think that clowns breaking ATMs for small amounts of cash is funny

I'd like some feedback regarding the change and also regarding my code!! I bugtested it a decent amount but tell me if I'm missing something!

[FEAT] [INPUT]

## Changelog

```
(u)Flaborized:
(*)You can now destroy chemical dispensers, ATMs and shrubs by smashing them with a weapon. ATMs will spew out small amounts of money if you do this, and can also be shot to destroy them! Also improved visual feedback when destroying fabricators through brute force.
```

### CODE REVIEW

**Comment by Adharainspace** [comment]
this is good!

**Comment by Gannets** [comment]
I really like the idea of more destructible/reactive items.

**Comment by mybluecorners** [comment]
do these things also take damage from ex_act? (i think they should)

**Comment by Flaborized** [comment]
> do these things also take damage from ex_act? (i think they should)

it doesnt but i can and will make it (later)

**Comment by ZeWaka** [comment]
> 
> 
> > do these things also take damage from ex_act? (i think they should)
> 
> it doesnt but i can and will make it (later)

clarification: would that in this PR or in a future one, for merging purposes

**Comment by Flaborized** [comment]
> > > do these things also take damage from ex_act? (i think they should)
> > 
> > 
> > it doesnt but i can and will make it (later)
> 
> clarification: would that in this PR or in a future one, for merging purposes

This one, I'll add this behavior in a bit Hopefully! Don't merge until I do this TIA

**Comment by Flaborized** [comment]
Made them lose health on ex_act, explosions should still destroy them if they would have before, or if they're weaker they'll do a portion of their health instead of rolling a chance to destroy them. Shrubs still get qdel-d if there's a big explosion cuz I thought that it was fitting for there to be "nothing left" instead of leaves left in that case

File: code/datums/banking.dm [hunk 1]
```diff
@@ -855,6 +881,28 @@
...
+					return
+				src.broken = 1
+				src.visible_message("<span style=\"color:red\"><b>The [src.name] breaks apart and spews out cash!</b></span>")
+				src.icon_state = "[src.icon_state]_broken"
+				var/obj/item/C = pick("/obj/item/spacecash/hundred", "/obj/item/spacecash/fifty", "/obj/item/spacecash/ten")
+				C = new C(get_turf(src))
+				playsound(src.loc,'sound/impact_sounds/Machinery_Break_1.ogg', 50, 2)
+				playsound(src.loc,'sound/machines/capsulebuy.ogg', 50, 2)
+				C.throw_at(usr, 20, 3)
+				return
```
Comment (flrsh) [inline code/datums/banking.dm:901]:
```suggestion
		if (src.health <= 0)
			src.broken = 1
			src.visible_message("<span style=\"color:red\"><b>The [src.name] breaks apart and spews out cash!</b></span>")
			src.icon_state = "[src.icon_state]_broken"
			var/obj/item/C = pick(/obj/item/spacecash/hundred, /obj/item/spacecash/fifty, /obj/item/spacecash/ten)
			C = new C(get_turf(src))
			playsound(src.loc,'sound/impact_sounds/Machinery_Break_1.ogg', 50, 2)
			playsound(src.loc,'sound/machines/capsulebuy.ogg', 50, 2)
			C.throw_at(user, 20, 3)
```

File: code/modules/chemistry/Chemistry-Dispenser.dm [hunk 2]
```diff
@@ -529,6 +539,22 @@
 			boutput(usr, "<span style=\"color:red\">You can't use that as an output target.</span>")
 		return
 
+	proc/take_damage(var/damage_amount = 0)
+		if (!damage_amount)
+			return
```
Comment (flrsh) [inline code/modules/chemistry/Chemistry-Dispenser.dm:544]:
```suggestion
	proc/take_damage(var/damage_amount = 5)
```

File: code/modules/chemistry/Chemistry-Dispenser.dm [hunk 3]
```diff
@@ -529,6 +539,22 @@
...
+		src.health -= damage_amount
+		src.health = max(0,min(src.health,100))
```
Comment (flrsh) [inline code/modules/chemistry/Chemistry-Dispenser.dm:546]:
```suggestion
```

File: code/modules/chemistry/Chemistry-Dispenser.dm [hunk 4]
```diff
@@ -529,6 +539,22 @@
...
+		if (damage_amount > 0)
+			if (src.health == 0)
+				if (beaker)
+					beaker:set_loc(src.output_target ? src.output_target : get_turf(src))
+					beaker = null
+				src.visible_message("<span style=\"color:red\"><b>The [src.name] falls apart into useless debris!</b></span>")
+				robogibs(src.loc,null)
+				playsound(src.loc,'sound/impact_sounds/Machinery_Break_1.ogg', 50, 2)
+				qdel(src)
+				return
```
Comment (flrsh) [inline code/modules/chemistry/Chemistry-Dispenser.dm:556]:
```suggestion
		if (src.health <= 0)
			if (beaker)
				beaker.set_loc(src.output_target ? src.output_target : get_turf(src))
				beaker = null
			src.visible_message("<span style=\"color:red\"><b>[name] falls apart into useless debris!</b></span>")
			robogibs(src.loc,null)
			playsound(src.loc,'sound/impact_sounds/Machinery_Break_1.ogg', 50, 2)
			qdel(src)
			return
```

File: code/obj/decorations.dm [hunk 5]
```diff
@@ -194,6 +191,27 @@
...
 		interact_particle(user,src)
 
+	attackby(var/obj/item/W as obj, user as mob)
+		var/damage = W.force
+		usr.lastattacked = src
+		hit_twitch(src)
+		attack_particle(user,src)
+		playsound(src, "sound/impact_sounds/Bush_Hit.ogg", 50, 1, 0)
+		src.take_damage(damage)
+		usr.visible_message("<span style='color:red'><b>[user] hacks at the [src] with [W]!</b></span>")
```
Comment (flrsh) [inline code/obj/decorations.dm:201]:
```suggestion
		user.lastattacked = src
		hit_twitch(src)
		attack_particle(user,src)
		playsound(src, "sound/impact_sounds/Bush_Hit.ogg", 50, 1, 0)
		src.take_damage(W.force)
		user.visible_message("<span style='color:red'><b>[user] hacks at [src] with [W]!</b></span>")
```
I think `[src]` will automatically include a "the" if it's lowercase.

File: code/obj/decorations.dm [hunk 6]
```diff
@@ -194,6 +191,27 @@
...
+		src.health -= damage_amount
+		src.health = max(0,min(src.health,100))
+		if (damage_amount > 0)
+			if (src.health == 0)
+				src.visible_message("<span style=\"color:red\"><b>The [src.name] falls apart!</b></span>")
+				var/obj/item/L = "/obj/decal/cleanable/leaves"
+				L = new L(get_turf(src))
+				playsound(src.loc, "sound/impact_sounds/Slimy_Hit_3.ogg", 100, 0)
+				qdel(src)
+				return
```
Comment (flrsh) [inline code/obj/decorations.dm:213]:
```suggestion
	proc/take_damage(var/damage_amount = 5)
		src.health -= damage_amount
		if (src.health <= 0)
			src.visible_message("<span style=\"color:red\"><b>The [src.name] falls apart!</b></span>")
			new /obj/decal/cleanable/leaves(get_turf(src))
			playsound(src.loc, "sound/impact_sounds/Slimy_Hit_3.ogg", 100, 0)
			qdel(src)
			return
```

File: code/datums/banking.dm [hunk 7]
```diff
@@ -881,24 +878,19 @@
...
-				src.visible_message("<span style=\"color:red\"><b>The [src.name] breaks apart and spews out cash!</b></span>")
-				src.icon_state = "[src.icon_state]_broken"
-				var/obj/item/C = pick("/obj/item/spacecash/hundred", "/obj/item/spacecash/fifty", "/obj/item/spacecash/ten")
-				C = new C(get_turf(src))
-				playsound(src.loc,'sound/impact_sounds/Machinery_Break_1.ogg', 50, 2)
-				playsound(src.loc,'sound/machines/capsulebuy.ogg', 50, 2)
-				C.throw_at(usr, 20, 3)
+	proc/take_damage(var/damage_amount = 5)
+		if (broken)
 				return
```
Comment (flrsh) [inline code/datums/banking.dm]:
```suggestion
		if (broken)
			return
```

Comment (flrsh) [reply code/datums/banking.dm]: Sorry! D:

**Pull request merged**